pkg/model/application_live_state.go

```diff
@@ -144,3 +144,37 @@ func (s *ApplicationLiveStateSnapshot) determineLambdaAppHealthStatus() {
 	}
 	s.HealthStatus = ApplicationLiveStateSnapshot_HEALTHY
 }
+
+// DetermineApplicationHealthStatus updates the health status of the application based on the health status of its resources.
+func (s *ApplicationLiveStateSnapshot) DetermineApplicationHealthStatus() {
+	app := s.ApplicationLiveState
+	if app == nil {
+		return
+	}
+
+	s.HealthStatus = ApplicationLiveStateSnapshot_HEALTHY
```

Comment:
Move this to L169, to make reading scope smaller.

Reply:
Fixed at [400e581](https://github.com/pipe-cd/pipecd/pull/5454/commits/400e581a7f7d0922a1e0a738cc4eff8cacc1361a)